crates/kubelet/src/module_store.rs

```diff
@@ -97,39 +202,83 @@ impl<C> FileModuleStore<C> {
     fn pull_path(&self, r: &Reference) -> PathBuf {
         let mut path = self.root_dir.join(r.registry());
         path.push(r.repository());
-        path.push(r.tag());
+        path.push(r.tag().unwrap_or("latest"));
         path
     }
 
     fn pull_file_path(&self, r: &Reference) -> PathBuf {
         self.pull_path(r).join("module.wasm")
     }
 
-    async fn store(&self, image_ref: &Reference, contents: &[u8]) -> anyhow::Result<()> {
+    fn digest_file_path(&self, r: &Reference) -> PathBuf {
+        self.pull_path(r).join("digest.txt")
```

Comment:
Isn't the digest some sort of SHA? Can we recalculate it instead of storing locally? I feel like you could do some weird fiddling if you can modify the digest file

Reply:
I thought we were means to use the digest from the registry.

And yeah, if someone fiddles with the internals of the store, then they could mess things up.  But if we're worried about that then shouldn't we also worry about someone taking a hex editor to the `module.wasm` file?

Reply:
Both are viable attacks, but the digest feels pretty simple to change. As for the digest, we do get it from the registry, but can't we calculate it locally as well? I honestly don't know enough here

Reply:
From my memory of the CNAB distribution wars, different registries can calculate digests differently (and differently from a client).  I am pretty sure one of the difficulties with that spec was that we could not get a digest for a bundle without first pushing it to a registry.

Reply:
Ah ok that makes sense